### PR TITLE
Medatagrapher fix

### PR DESCRIPTION
**Pull request by Nono1971**
Hi Ocramius,

Well... after My last pull request, I created the test as you asked... and the testCase found another bug :) 
=> depending of metadata's order, there was another inconcistency : 
If the inversed side associations were visited first, we had a double arrow for the relation...

Fixed it too :)

I must tell you that I took inspiration from your YumlController to create a Symfony bundle... I Missed yUML diagram so much when I started using Doctrine with CodeIgniter (first) and now with Symfony....

https://packagist.org/packages/onurb/doctrine-yuml-bundle

I first reused in my bundle your MetadataGrapher class as is (only changed the namespace) : I didn't want to load DoctrineORMModule and all its dependencies for only one class !... (the only problem for me is now to write code twice to push you my fixes ;) )

I think it's fair to inform you that you're work has been reused :)

Thank's Marco for you're work on Doctrine (and all the rest of the team of course)... I just love this library !.

Bruno.

### CODE REVIEW

**Comment by Ocramius** [comment]
> I think it's fair to inform you that you're work has been reused :)

If you extract it to a separate lib, we can require it from here and from the bundles ;-)

**Comment by Ocramius** [comment]
@gianarb this looks good: please make sure that you are happy with the tests before merging.

File: tests/DoctrineORMModuleTest/Yuml/MetadataGrapherTest.php [hunk 1]
```diff
@@ -400,4 +400,74 @@ function ($assoc) {
             $this->grapher->generateFromMetadata(array($class1, $class2))
         );
     }
+
+    /**
+     * @covers \DoctrineORMModule\Yuml\MetadataGrapher
+     */
+    public function testBugTestCaseFix()
```
Comment (Ocramius) [inline tests/DoctrineORMModuleTest/Yuml/MetadataGrapherTest.php]: This test verifies multiple cases - please consider chunking it into smaller tests explaining each scenario carefully in the method name (such as `testAssociationOneToManyBlaBlaDoesDoBlaBla`)

**Comment by gianarb** [comment]
Yes sorry for the last merge without test :) my slip :+1:

**Comment by Nono1971** [comment]
Infact, I have 6 assertions in the test, but the Case is the same each time : three classes, two relations on the same class with the same name, one bidirectional and one not...

The only difference between the  assertions is the order of the ClassMetadata send to the MetadataGrapher... I think I could make a dataprovider to inject cases in the test... 

I work on it and push when ready ! For the method name... I was not inspired... I'll do better ;)


I like your idea to separate the MetadataGrapher as a specific library :) I'll do it when we'll agree the code here :) (

**Comment by Nono1971** [comment]
Tell me if you prefer tests like this ^^

**Comment by Nono1971** [comment]
oups, I have a problem... 
My test pass when I launch it individualy... but another test fails now ^^: tests\DoctrineORMModuleTest\Service\ConfigurationFactoryTest.php:304
I checked... I have the same failure on my master branch... wich is up to date with upstream ^^ so there is nothing I think I can do for it ^^

Scrutinizer also failed "The code coverage data was not received within the specified time. Please make sure your third-party service is configured to send code coverage data."... can we relaunch it ?

Can you check test manualy to validate the commit ? or is there anything I still can do by my side ?

**Comment by Nono1971** [comment]
Try to push after merging master changes... 
finger crossed !

**Comment by Ocramius** [comment]
@gianarb this will need a rebase, eventually

**Comment by Ocramius** [comment]
@Nono1971 test looks much better, thanks :)

**Comment by Nono1971** [comment]
Just refactored the MetadataGrapher... 
But there is still a failure in unit tests... 
I really don't figure how my changes in MetadataGrapher and associated tests could break anything somewhere else... it isn't used anywhere but in the YumlController !

**Comment by Nono1971** [comment]
Yes !! I just saw that Gianluca fixed it in #433
I'll merge and push back my PR when merged in master :)

**Comment by gianarb** [comment]
I rebased and fixed Cs bug here #435 

Thanks!

**Comment by Nono1971** [comment]
Great ! thanks you too !

**Pull request closed without merging**